openssl-dynamic/src/main/java/io/netty/internal/tcnative/NativeStaticallyReferencedJniMethods.java

```diff
@@ -0,0 +1,130 @@
+package io.netty.internal.tcnative;
+
+/**
+ * This class is necessary to break the following cyclic dependency:
+ * <ol>
+ * <li>JNI_OnLoad</li>
+ * <li>JNI Calls FindClass because RegisterNatives (used to register JNI methods) requires a class</li>
+ * <li>FindClass loads the class, but static members variables of that class attempt to call a JNI method which has not
+ * yet been registered.</li>
+ * <li>java.lang.UnsatisfiedLinkError is thrown because native method has not yet been registered.</li>
+ * </ol>
+ * Static members which call JNI methods must not be declared in this class!
+ */
+final class NativeStaticallyReferencedJniMethods {
```

Comment:
I don't have a strong preference either way but just to explain the rational for the existing name:

- To clarify that not all constants have to go here (only the statically referenced ones), and that this class exists specifically to avoid class loader issues. So folks should not add things to this class which may introduce a class loader cycle described in the class comments. For example someone may think "I will also add a `static final int MY_VAR = someJNIMethod();` in this class to make the value easier to access from java." which would be wrong.
- To be consistent with transport-native-epoll. The plan is to move tcnative into the Netty project so consistently will help folks understand similar concepts across module boundaries.

Also this class is internal so folks will not be exposed to it.